### PR TITLE
feat: on Price create, create (or get) Location, and link them

### PR DESCRIPTION
**Pull request by raphodn**
### What

- new relationship `Price.location`
- new background task `create_price_location`
- on Price create, we first create (or get if existing) a Location, and then link them

### CODE REVIEW

File: app/tasks.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from sqlalchemy.orm import Session
+
+from app import crud
+from app.schemas import LocationCreate, PriceBase
+
+
+def create_price_location(db: Session, price: PriceBase):
```
Comment (raphael0202) [inline app/tasks.py:7]: are `Session` objects thread-safe? As I assume we're using threads to launch a task in the background.

**Review by raphael0202 — APPROVED** [review]

File: app/tasks.py (hunk 1 above)
Comment (raphodn) [reply app/tasks.py:7]: No idea 😅
I used the background tasks example from the fastAPI tutorials..

Comment (raphodn) [reply app/tasks.py:7]: though they do recommend to create a new db session here: https://github.com/tiangolo/fastapi/discussions/8502#discussioncomment-5606921

Comment (raphodn) [reply app/tasks.py:7]: added it to #31

**Pull request merged**